test/test_core/test_game_entity.py

```diff
@@ -0,0 +1,49 @@
+import unittest
+from unittest.mock import patch
+import logging
+from pyndemic.core.context import ContextRegistrationMeta, ContextError
+from pyndemic.core import api
+from queue import Queue
+import warnings
+
+from pyndemic.core.game_entity import GameEntity
+
+
+class DumbQueue:
+    def __init__(self):
+        self.queue = list()
+
+    def put(self, something):
+        self.queue.append(something)
+
+    def get(self):
+        return self.queue.pop(0)
+
+
+class MockController(metaclass=ContextRegistrationMeta,
+                     ctx_name='controller'):
+    def __init__(self):
+        self.signals = DumbQueue()
+
+    def create_entity(self):
+        return GameEntity()
+
+
+class GameEntityTestCase(unittest.TestCase):
+    def setUp(self):
+        self.controller = MockController()
+
+    def test_assert_has_context(self):
+        with self.assertRaises(ContextError):
+            with warnings.catch_warnings(): # catch does not work
+                warnings.simplefilter('ignore')
```

Comment:
Which warnings suppressed here?
I couldn't find any in the project code.

Reply:
GameEntity is created here without context, so GameEntityCreationMeta logs a warning. It appears only when I run test only in this file. When I run the whole suit it shows nothing. Perhaps this is so because it is a log, not throw.